packages/components/src/form/test/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { createElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Form from '../';
+
+// There is async code inside the callback handlers that we must flush
+// before asserting they were called.
+function flushPromises() {
+	return new Promise( function ( resolve ) {
+		setTimeout( resolve );
+	} );
+}
+
+describe( 'Form', () => {
+	it( 'should default to call the deprecated onSubmitCallback if it is provided.', async () => {
+		const mockOnSubmitCallback = jest.fn();
+		const mockOnSubmit = jest.fn();
+
+		const { queryByText } = render(
+			<Form
+				onSubmitCallback={ mockOnSubmitCallback }
+				onSubmit={ mockOnSubmit }
+				validate={ () => ( {} ) }
+			>
+				{ ( { handleSubmit } ) => {
+					return <button onClick={ handleSubmit }>Submit</button>;
+				} }
+			</Form>
+		);
+
+		userEvent.click( queryByText( 'Submit' ) );
+
+		await flushPromises();
+		await expect( mockOnSubmitCallback ).toHaveBeenCalledTimes( 1 );
+		await expect( mockOnSubmit ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should default to call the deprecated onChangeCallback prop if it is provided.', async () => {
+		const mockOnChangeCallback = jest.fn();
+		const mockOnChange = jest.fn();
+
+		const { queryByText } = render(
+			<Form
+				onChangeCallback={ mockOnChangeCallback }
+				onChange={ mockOnChange }
+				validate={ () => ( {} ) }
+			>
+				{ ( { setValue } ) => {
+					return (
+						<button
+							onClick={ () => {
+								setValue( 'foo', 'bar' );
+							} }
+						>
+							Submit
+						</button>
+					);
+				} }
+			</Form>
+		);
+
+		userEvent.click( queryByText( 'Submit' ) );
```

Comment:
I'd rename it to "Change" not to suggest that it's a submit button.
```suggestion
							Change
						</button>
					);
				} }
			</Form>
		);

		userEvent.click( queryByText( 'Change' ) );
```

Reply:
Sure happy to change that.